modules/gapi/src/api/gmat.cpp

```diff
@@ -36,37 +36,26 @@ const cv::GOrigin& cv::GMat::priv() const
     return *m_priv;
 }
 
-int cv::checkVector(const cv::GMatDesc& in, const int n, const int ddepth)
-{
-    GAPI_Assert(ddepth == in.depth || ddepth < 0);
-    const int chan = in.chan;
-    const cv::Size size = in.size;
-    int quantity = -1;
-    if (n <= 0)
-    {
-        if (size.width == 1 || chan == 1)
-        {
-            quantity = size.height;
-        }
-        else if (size.height == 1)
-        {
-            quantity = size.width;
-        }
-        // else input Mat can't be described as vector of points
-    }
-    else // n is given
-    {
-        if (chan == n && size.height == 1)
-        {
-            quantity = size.width;
-        }
-        else if ((chan == n && size.width == 1) || (chan == 1 && size.width == n))
-        {
-            quantity = size.height;
-        }
-        // else input Mat can't be described as vector of n-dimentional points
-    }
-    return quantity;
+std::tuple<int,int> cv::checkVector(const cv::GMatDesc& in, const int n, const int expectedDepth)
```

Comment:
No, no, why ``cv::``?
``cv::checkVector`` which works only for ``cv::GMatDesc``?

Reply:
well, yes, this `checkVector` of mine, it is in the `cv::` namespace

I guess you think it can shadow the `checkVector` of `cv::Mat`, but it is actually `cv::Mat::checkVector`, its method

Reply:
Its namespace belonging is similar to `cv::descr_of`, for example (right below)

Reply:
No, I just meant that user will see the `cv::checkVector` method but don't see that this is only for G-API type.
`cv::descr_of` has more speaking name..

Reply:
I see. I moved this function to cv::gapi:: namespace.
@alalek Is it a good decision? And do we need tests for this function as it's a part of the public API?

Reply:
moved further, in `cv::gapi::detail`